sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -36,7 +36,7 @@ case class PrecisionInfo(precision: Int, scale: Int) {
 /**
  * :: DeveloperApi ::
  * The data type representing `java.math.BigDecimal` values.
- * A Decimal that might have fixed precision and scale, or unlimited values for these.
+ * A Decimal that must have fixed precision and scale.
  *
  * Please use [[DataTypes.createDecimalType()]] to create a specific instance.
  */
```

Comment:
would it make more sense to make precision and scale the two parameters of DecimalType, rather than an optional PrecisionInfo? 

To keep binary compatibility, we can still create a function called precisionInfo